spec/cube_spec.rb

```diff
@@ -436,4 +436,171 @@
 
   end
 
+  describe "cache" do
+    before(:all) do
+      @connection = ActiveRecord::Base.connection
+      @cube = @olap.cube('Sales')
+      @query = <<-SQL
+        SELECT {[Measures].[Store Cost], [Measures].[Store Sales]} ON COLUMNS
+        FROM  [Sales]
+        WHERE ([Time].[2010].[Q1], [Customers].[USA].[CA])
+      SQL
+    end
+
+    after(:each) do
+      Rake::Task['db:create_sales_data'].reenable
+      Rake::Task['db:create_sales_data'].invoke
```

Comment:
Instead of regenerating data using the Rake task I suggest to

* in `before(:all)` create a temp table `sales_copy` as `select * from sales` (might be necessary to write different SQL based on the database server)
* in `after(:each)` truncate `sales` and insert `select * from sales_copy`
* in `after(:all)` drop table `sales_copy`

It should be faster and then it will not be necessary to load rake tasks here.